test/app-luatest/gh_5444_relax_uuid_validation_test.lua

```diff
@@ -0,0 +1,48 @@
+local uuid = require('uuid')
+local msgpack = require('msgpack')
+local t = require('luatest')
+
+local g = t.group()
+
+-- Base test.
+g.test_all_versions = function()
+    local values = {
+        -- UUID values generated with https://www.uuidtools.com/.
+        -- As a namespace ns.URL:tarantool.org is used.
+        v1 = '7ff4307c-48f1-11ee-ac1a-325096b39f47',
+        v2 = '000003e8-48f1-21ee-b000-325096b39f47',
+        v3 = 'da8e829c-c63e-33df-af97-359b96647e2d',
+        v4 = 'ff4b9fc2-620e-4e5c-a1b9-ff241c4793f0',
+        v5 = 'a658ebe8-3c03-566d-bb9b-069247787734',
+        -- UUID examples for new versions taken from the IETF
+        -- draft:
+        -- https://www.ietf.org/archive/id/draft-peabody-dispatch-new-uuid-format-04.html.
+        v6 = '1ec9414c-232a-6b00-b3c8-9e6bdeced846',
+        v7 = '017f22e2-79b0-7cc3-98c4-dc0c0c07398f',
+        v8 = '320c3d4d-cc00-875b-8ec9-32d5f69181c0',
+    }
+
+    for v, u in pairs(values) do
+        t.assert(uuid.fromstr(u),
+                 ('UUID %s value (%s) failed to parse'):format(v, u))
+    end
+end
```

Comment:
How about using [test vectors](https://datatracker.ietf.org/doc/html/rfc9562#name-test-vectors) from the final RFC 9562 edition?

The online tool says just 'Sorry, you have been blocked' for me. (And I don't understand what is the mentioned namespace and how it affects the generated UUID values.)

The final version of the RFC sounds better than its draft.

Reply:
Replaced with the test vectors, as you suggested. Since implementation of v2 isn't defined, it is skipped in the test. OTOH, time-based and name-based versions of v8 UUID are checked.